yarn.config.cjs

```diff
@@ -0,0 +1,299 @@
+      'The README.md contains template instructions. These instructions should be removed.',
+    );
+  }
+
+  if (!readme.includes(`yarn add ${packageName}`)) {
+    workspace.error(
+      `The README.md does not contain an example of how to install the package using Yarn (\`yarn add ${packageName}\`). Please add an example.`,
+    );
+  }
+
+  if (!readme.includes(`npm install ${packageName}`)) {
+    workspace.error(
+      `The README.md does not contain an example of how to install the package using npm (\`npm install ${packageName}\`). Please add an example.`,
+    );
+  }
+}
+
+/**
+ * Expect that the workspace has a pull_request_template.md file, and that it
+ * is a non-empty string. The pull_request_template.md is expected to:
+ *
+ * - Not contain an examples section (unless the workspace is the module
+ * template itself).
+ *
+ * @param {Workspace} workspace - The workspace to check.
+ * @param {string} workspaceName - The name of the workspace.
+ * @returns {Promise<void>}
+ */
+async function expectPullRequestTemplate(workspace, workspaceName) {
+  if (workspaceName === 'metamask-module-template') {
+    return;
+  }
+
+  const pullRequestTemplate = await getWorkspaceFile(
+    workspace,
+    '.github/pull_request_template.md',
+  );
+
+  if (!pullRequestTemplate) {
+    workspace.error(
+      'The pull_request_template.md is missing. This should be added.',
+    );
+  }
+
+  if (pullRequestTemplate.includes('## Examples')) {
+    workspace.error(
+      'The pull_request_template.md contains an examples section. This section should be removed.',
+    );
+  }
+}
+
+/**
+ * Expect that the workspace has a valid `exports` field. The `exports` field
+ * is expected to:
+ *
+ * - Export a `types` entrypoint as the first export, or not at all.
+ *
+ * This is required for proper TypeScript support when using `Node16` (or later)
+ * module resolution.
+ *
+ * @param {Workspace} workspace - The workspace to check.
+ * @returns {void}
+ */
+function expectExports(workspace) {
+  const { exports: manifestExports } = workspace.manifest;
+
+  if (!manifestExports) {
+    workspace.error("There are no exports defined in package.json.");
+    return;
+  }
+
+  Object.entries(manifestExports)
+    .filter(([, exportValue]) => typeof exportValue !== 'string')
+    .forEach(([exportName, exportObject]) => {
+      const keys = Object.keys(exportObject);
+      if (keys.includes('types') && keys[0] !== 'types') {
+        workspace.error(
+          `The "types" export must be the first export in the "exports" field for the export "${exportName}".`,
+        );
+      }
+    });
+}
+
+module.exports = defineConfig({
+  async constraints({ Yarn }) {
+    const workspace = Yarn.workspace();
+    const workspaceName = getWorkspaceName(workspace);
+    const workspaceRepository = `${BASE_URL}${workspaceName}`;
+
+    // The package must have a name, version, description, and license.
+    // TODO: Update when we rename the package
```

Comment:
I've adjusted the constraints file to exclude certain checks that are in the module template as we would need to make some other changes such as bumping the Node version for this to fully pass. We can progressively re-enable these checks as we make those changes.